packages/discussions/src/types.ts

```diff
@@ -0,0 +1,839 @@
+  status?: PostStatus[];
+  relations?: PostRelation[];
+}
+
+/**
+ * dto for querying posts in many channels
+ *
+ * @export
+ * @interface IQueryPosts
+ * @extends {IQueryChannelPosts}
+ */
+export interface IQueryPosts extends IQueryChannelPosts {
+  groups?: string[];
+  access?: SharingAccess[];
+}
+
+/**
+ * dto for updating a post's channel
+ *
+ * @export
+ * @interface IUpdatePostSharing
+ * @extends {Partial<IWithSharing>}
+ */
+export interface IUpdatePostSharing extends Partial<IWithSharing> {}
+
+/**
+ * dto for updating a post's status
+ *
+ * @export
+ * @interface IUpdatePostStatus
+ */
+export interface IUpdatePostStatus {
+  status: PostStatus;
+}
+
+/**
+ * dto for updating a post's content
+ *
+ * @export
+ * @interface IUpdatePost
+ */
+export interface IUpdatePost {
+  title?: string;
+  body?: string;
+}
+
+// // channels
+/**
+ * representation of channel entity
+ *
+ * @export
+ * @interface IChannel
+ * @extends {IWithSettings}
+ * @extends {IWithSharing}
+ * @extends {IWithAuthor}
+ * @extends {IWithTimestamps}
+ */
+export interface IChannel
+  extends IWithSettings,
+    IWithSharing,
+    IWithAuthor,
+    IWithTimestamps {
+  id: number;
+}
+
+/**
+ * dto for creating a channel
+ *
+ * @export
+ * @interface ICreateChannel
+ * @extends {IWithSettings}
+ * @extends {IWithSharing}
+ */
+export interface ICreateChannel extends IWithSettings, IWithSharing {}
+
+/**
+ * dto for decorating found channel with relations
+ *
+ * @export
+ * @interface IGetChannel
+ */
+export interface IGetChannel {
+  relations?: ChannelRelation[];
+}
+
+/**
+ * dto for querying channels
+ *
+ * @export
+ * @interface IQueryChannels
+ * @extends {Partial<IPagingParams>}
+ * @extends {Partial<IWithSorting>}
+ * @extends {Partial<IWithTimeQueries>}
+ */
+export interface IQueryChannels
+  extends Partial<IPagingParams>,
+    Partial<IWithSorting>,
+    Partial<IWithTimeQueries> {
+  groups?: string[];
+  access?: SharingAccess[];
+  relations?: ChannelRelation[];
+}
+
+/**
+ * dto for updating channel settings
+ *
+ * @export
+ * @interface IUpdateChannel
+ * @extends {Partial<IWithSettings>}
+ */
+export interface IUpdateChannel extends Partial<IWithSettings> {}
+
+// // reactions
+
+/**
+ * representation of reaction entity
+ *
+ * @export
+ * @interface IReaction
+ * @extends {IWithAuthor}
+ * @extends {IWithTimestamps}
+ */
+export interface IReaction extends IWithAuthor, IWithTimestamps {
+  id: number;
+  value: PostReaction;
+  postId?: number;
+  post?: IPost;
+}
+
+/**
+ * dto for creating a reaction
+ *
+ * @export
+ * @interface ICreateReaction
+ */
+export interface ICreateReaction {
+  value: PostReaction;
+}
+
+// request options
+
+/**
+ * options for making requests against Discussion API
+ *
+ * @export
+ * @interface IRequestOptions
+ * @extends {RequestInit}
+ */
+export interface IRequestOptions extends RequestInit {
+  authentication?: IAuthenticationManager;
+  token?: string;
+  apiBaseUrl?: string;
+  params?: {
```

Comment:
This works different than https://esri.github.io/arcgis-rest-js/api/request/IRequestOptions/#params which is just a set of key/value pairs that are turned into query or body params based on the method.